drizzlepac/haputils/product.py

```diff
@@ -299,67 +299,102 @@ def add_member(self, edp):
+            crclean.append(edp.crclean)
+
+        # These should be in a configuration file
+        catalog_list = [catalog_name, 'GAIADR2']
+        num_cat = len(catalog_list)
+        method_list = ['relative', 'default']
+
         try:
+            # Proceed only if there is data to process
             if self.edp_list:
-                for edp in self.edp_list:
-                    exposure_filenames.append(edp.full_filename)
-                    headerlet_filenames[edp.full_filename] = edp.headerlet_filename
-                    crclean.append(edp.crclean)
 
+                # If necessary, generate the alignment table only once
                 if align_table is None:
                     align_table = align_utils.AlignmentTable(exposure_filenames,
                                                              log_level=self.log_level,
                                                              **alignment_pars)
 
                     align_table.find_alignment_sources(output=output, crclean=crclean)
                     align_table.configure_fit()
-                log.debug('Creating reference catalog {}'.format(self.refname))
-                ref_catalog = amutils.create_astrometric_catalog(align_table.process_list,
-                                                                 catalog=catalog_name,
-                                                                 output=self.refname,
-                                                                 gaia_only=False)
-
-                log.debug("Abbreviated reference catalog displayed below\n{}".format(ref_catalog))
-                align_table.reference_catalogs[self.refname] = ref_catalog
-                if len(ref_catalog) > align_utils.MIN_CATALOG_THRESHOLD:
-                    fit_again = False
-                    try:
-                        align_table.perform_fit(method_name, catalog_name, ref_catalog,
-                                                fitgeom=fitgeom)
-                    except Exception:
-                        log.info('Problem with initial fit...')
-                        fit_again = True
-
-                    # If there was a problem with the first fit...
-                    if fit_again:
-                        log.info("Trying 'DEFAULT' fit to {} instead.".format(catalog_name))
-                        # Try one more time with slightly different cross-matching algorithm
-                        method_name = 'default'
-                        align_table.configure_fit()
-                        align_table.perform_fit(method_name, catalog_name, ref_catalog,
-                                                fitgeom=fitgeom)
 
-                    align_table.select_fit(catalog_name, method_name)
+                is_good_fit = False
+                for index_cat, catalog_item in enumerate(catalog_list):
+                    log.debug('Creating reference catalog {}'.format(self.refname))
```

Comment:
This reference filename will get overwritten by the second catalog if the first one fails for any reason.  The first catalog would therefore not be available to understand what went wrong in case it was a problem with the ref cat itself (too few sources, sources too close together, sources not the ones identified in the image, ...).

Reply:
Every reference catalog now has the catalog name as part of the filename.